### PR TITLE
ZF3 Compatibility

### PR DESCRIPTION
**Pull request by webimpress**
Added support for ZF3 modules. Only PHP5.6 and PHP7.0 are supported.
Depends on:
* ~~https://github.com/doctrine/DoctrineModule/pull/567~~ [DoctrineModule 1.2.0](https://github.com/doctrine/DoctrineModule/releases/tag/1.2.0)
* ~~https://github.com/doctrine/DoctrineMongoODMModule/pull/168~~ [DoctrineMongoODMModule 0.11.0](https://github.com/doctrine/DoctrineMongoODMModule/releases/tag/0.11.0)
* ~~https://github.com/doctrine/DoctrineORMModule/pull/493~~ [DoctrineORMModule 1.1.0](https://github.com/doctrine/DoctrineORMModule/releases/tag/1.1.0)

### CODE REVIEW

**Comment by veewee** [comment]
Hi @webimpress,

Thanks for the PR, it looks pretty good!
The only thing I am worried about is the amount of non official modules and it's versions:

```
"api-skeletons/zf-doctrine-module-zend-hydrator": "^1.0",
"doctrine/doctrine-module": "dev-feature/zf3compat as 1.0",
 "doctrine/doctrine-orm-module": "dev-feature/zf3compat",
"doctrine/doctrine-mongo-odm-module": "dev-feature/zf3compat",
```

Do you have any idea on what's the state on ZF3 compatibility in the doctrine modules?
As long as those projects aren't ZF3 compatible, it might be better to wait until the features are merged in the officiel repo?

**Comment by webimpress** [comment]
ping @veewee 

All dependencies are now tagged. Please release when you have a moment. Thanks!

File: composer.json [hunk 1]
```diff
@@ -1,40 +1,47 @@
...
+    "doctrine/mongodb-odm": "^1.1",
+    "phpro/grumphp": "^0.9.1"
+  },
+  "autoload": {
+    "psr-4": {
+      "Phpro\\DoctrineHydrationModule\\": "src/"
+    }
+  },
+  "autoload-dev": {
+    "PhproTest\\DoctrineHydrationModule\\": "test/src/"
```
Comment (webimpress) [inline composer.json]: Good catch! Thanks. Changed 8aba9327e60f50de4d77316fd4b7b73bff684fe5

File: composer.json [hunk 2]
```diff
@@ -1,40 +1,47 @@
...
+    "zendframework/zend-servicemanager": "^2.7.6 || ^3.1",
+    "zendframework/zend-hydrator": "^1.1 || ^2.2.1",
+    "api-skeletons/zf-doctrine-module-zend-hydrator": "^1.0",
+    "doctrine/doctrine-module": "^1.2",
+    "doctrine/instantiator": "^1.0.5"
+  },
+  "require-dev": {
+    "fabpot/PHP-CS-Fixer": "^1.11.6",
+    "phpunit/phpunit": "^4.8",
+    "zendframework/zend-modulemanager": "^2.7.2",
```
Comment (webimpress) [inline composer.json]: Moved from `require-dev` to `require` section in composer 939f9c8cfc7d1f08b7b918947764877a737b8666

**Comment by webimpress** [comment]
@veewee thanks for review, please have a look on my fixes.

**Comment by veewee** [comment]
@webimpress: released in 3.0.0: https://github.com/phpro/zf-doctrine-hydration-module/releases/tag/v3.0.0

**Comment by webimpress** [comment]
@veewee thanks !

**Pull request merged**